deploy/config/crds/submarineraddon.open-cluster-management.io_submarinerconfigs.yaml

```diff
@@ -40,6 +40,10 @@ spec:
                 default: 4500
                 description: IPSecNATTPort represents IPsec NAT-T port (default 4500).
                 type: integer
+              NATTDiscoveryPort:
+                default: 4900
+                description: NATTDiscoveryPort represents NAT-T Discovery port (default UDP/4900).
```

Comment:
In a future this may need to be a list, for example for on-premise deployments, where each gateway (unless it hass a public IP of it's own) needs a separate port.

Do you think it could be doable to start using a list now?

Also IPSEC port has the same issue, so may be it's simpler to do it the way you do it now, and then in the future when we support on-prem better with this add-on you move both fields into lists.

Reply:
There is a way we can make it work without needing it to be a list. For multiple gateways we can treat configured port as base port and increment by 1 for every extra gateway. e.g. in config we only have 4500 for IPSEC, but if user configures 3 gateways we can use 4500, 4501 and 4502. Such a change can be easily implemented on submariner code without needing changes anywhere else.

If above doesn't work, we can always make it a user configured list later.